src/v/storage/segment_deduplication_utils.cc

```diff
@@ -227,7 +227,7 @@ ss::future<index_state> deduplicate_segment(
       &cmp_idx_writer,
       inject_reader_failure);
 
-    auto new_idx = co_await rdr.consume(
+    auto new_idx = co_await std::move(rdr).consume(
```

Comment:
nice. i just lost my foot looking at this.

I think this if fine, but i would leave a ticket to fix record_batch_reader::impl::consume, by chaining inside consume()& an then_wrapped() to clean up in case of exceptions

Reply:
we have append_entries_request::serde_async_write that does a consume like this, maybe it warrants a similar treatment

Reply:
or at least a comment around this in reader::consume about the exception safety